### PR TITLE
[Feature] Automatic class names based on config

### PR DESCRIPTION
**Pull request by ketsappi**
## Description

* Class names are now put automatically during the build of the icons library based on the given config.

* With the created plugin we can configure which attribute(s) to look for value(s) and then add the given class name for it based on those.

* Updated the icons from the designs; previous manually made changes to icons are lost.
* Changed illustrative icons svg class names. Fill and stroke are now combined and base color class name is added.
* svg’s in the repository no longer contain class names.

Currently only illustrative icons are gone through with the config.

## Details for config

To config the rules for the `addClassNamesPlugin`, rules are added in the `webpack.config.js`.

You can use regexp in the `attribute` and `value` properties to do more with one rule.

Here are some example rules:

```javascript
rules: [
  {
    attribute: 'fill',
    value: '#A5ADB1',
    className: 'fi-icon-illustrative-base'
  },
  {
    attribute: 'fill',
    value: '#E97025',
    className: 'fi-icon-illustrative-highlight'
  },
  {
    attribute: 'fill|stroke',
    value: '#FFFFFF|#FFF|#FFF000',
    className: 'fi-icon-illustrative-magical'
  },
]

```

## Motivation and context

No need to manually edit the icons exported from the designs to include class names.

## Testing

Tested to build and locally ran with the styleguidist by overriding colours by using the old&new class names.
Separate PR for `suomifi-ui-components` is made where these new class names can be used with old&new props of the icon component.

### Testing for PR reviewer

To test the changes, run the `yarn build` with the previous version and look `dist/index.js` in editor and compare it tothe version you build with this one. You can search for `className` or by using the color codes of `#A5ADB1` and `#E97025` to search for elements that has been affected with the change and having the classNames.

E.g search result:

```javascript
fill:"#A5ADB1",className:"fi-icon-illustrative-base"
```

## Links

Used the [removeAttrs](https://github.com/svg/svgo/blob/master/plugins/removeAttrs.js) as an example and guideline when writing the plugin.

## Release notes

* **Breaking change**: Change illustrative icons svg class names. Fill and stroke are now combined and base color class name is added.
* **Breaking change**: svg’s in the repository no longer contain class names.

### CODE REVIEW

File: src/illustrativeIcons/icon-illustration-family.svg [hunk 1]
```diff
@@ -1,9 +1,9 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <svg width="80px" height="80px" viewBox="0 0 80 80" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
```
Comment (aappoalander) [inline src/illustrativeIcons/icon-illustration-family.svg:2]: The sizes do not seem to be uniform. I think this should be reported to PAMU even if it does not make a difference in our case.

File: webpack.config.js [hunk 2]
```diff
@@ -32,7 +33,34 @@ module.exports = (env) => ({
...
+                  },
+                  {
+                    addClassNamesPlugin: Object.assign(
+                      {},
+                      addClassNamesPlugin,
+                      {
+                        params: {
+                          rules: [
+                            {
+                              attribute: 'fill',
```
Comment (aappoalander) [inline webpack.config.js]: I would configure the same classname to be applied to both, fill and stroke and to drop it from the name itself. This way we would have base and highlight classes per icon type.

**Comment by aappoalander** [comment]
I think these should be applied to all icon types, but this needs to be discussed.

File: webpack.config.js (hunk 2 above)
Comment (ketsappi) [reply webpack.config.js]: Is this like you meant?

```js
rules: [
  {
    attribute: 'fill|stroke',
    value: '#A5ADB1',
    className: 'fi-icon-illustrative-base'
  },
  {
    attribute: 'fill|stroke',
    value: '#E97025',
    className: 'fi-icon-illustrative-highlight'
  }
]

```

Comment (ketsappi) [reply webpack.config.js]: This rule change is now committed.

File: src/illustrativeIcons/icon-illustration-family.svg (hunk 1 above)
Comment (aappoalander) [reply src/illustrativeIcons/icon-illustration-family.svg:2]: This can also be left as maintenance task for later.

Comment (ketsappi) [reply src/illustrativeIcons/icon-illustration-family.svg:2]: Reported to PAMU.

**Review by aappoalander — APPROVED** [review]

**Pull request merged**